### DISCUSSION TITLE
[duke][Quectel] The protocol com.qualcomm.firehose does not allow the category to be X-Device.

### DISCUSSION THREAD

**Discussion by dukexinaw** [fwupd/fwupd#5357] [General]
We need to push the Quectel EM160 FW in Testing state to Stable state. LVFS services error warning was encountered.
: The protocol com.qualcomm.firehose does not allow the category to be X-Device.

 
XML Export
For future uploads this would be the contents of firmware.metainfo.xml:

<?xml version='1.0' encoding='utf-8'?>
<component type="firmware">
  <id>com.lenovo.quectel.em160.firmware</id>
  <name>QUECTEL.EM160</name>
  <summary>Firmware for the Quectel QUECTEL.EM160 modem</summary>
  <description>
    <p>Updating the firmware on your QUECTEL.EM160 device improves stability and adds new features.</p>
  </description>
  <provides>
    <firmware type="flashed">1e2db24f-667d-5a13-a69b-58a83eb5e4a5</firmware>
    <firmware type="flashed">235ba1e7-27da-554f-8ace-e041e7300f75</firmware>
    <firmware type="flashed">7c769da6-0d44-5b06-b917-0ff3de42baa8</firmware>
    <firmware type="flashed">f16f9c0e-ee5b-598d-b8ea-e0abe3b05d0f</firmware>
    <firmware type="flashed">f6cc7a2c-aa04-549a-9d8d-f9d2f571395e</firmware>
  </provides>
  <url type="homepage">https://www.quectel.com/support/</url>
  <metadata_license>CC0-1.0</metadata_license>
  <project_license>LicenseRef-proprietary</project_license>
  <categories>
    <category>X-Device</category>
  </categories>
  <custom>
    <value key="LVFS::VersionFormat">plain</value>
    <value key="LVFS::UpdateProtocol">com.qualcomm.firehose</value>
  </custom>
  <releases>
    <release version="EM160RGLAPR02A09M4G_07.007.07.007" date="2021-12-15" urgency="high">
      <checksum filename="firmware.zip" target="content"/>
      <description>
        <p>Firmware release EM160RGLAPR02A09M4G_07.007.07.007 packaged for LVFS.</p>
        <p>This is for Thinkcenter</p>
        <p>Version number: EM160RGLAPR02A09M4G_07.007.07.007</p>
        <p>This stable release fixes the following issues::</p>
        <ul>
          <li>performance issues</li>
        </ul>
      </description>
    </release>
  </releases>
  <requires>
    <id compare="ge" version="1.2.6">org.freedesktop.fwupd</id>
    <firmware compare="ge" version="EM160RGLAPR02A09M4G"/>
    <hardware>68d89030-a349-5667-be76-3ef7abff6323|c1180238-8ffa-5cbd-8532-3ee31f8e2e98</hardware>
  </requires>
  <keywords>
    <keyword>lenovo</keyword>
    <keyword>em160</keyword>
  </keywords>
</component>

**Comment by hughsie** [comment]
Would X-NetworkInterface be a better choice? Or could we create a X-WirelessDevice or X-Modem perhaps? I've changed the LVFS to allow X-Device for now, but we need to do something better in the medium term.

**Comment by dukexinaw** [comment]
Would X-NetworkInterface be a better choice? Or could we create a X-WirelessDevice or X-Modem perhaps? 
>>  got it, we will try this method.

I've changed the LVFS to allow X-Device for now, but we need to do something better in the medium term.
>>  Thank you very much. I reported to the customer that lvfs temporarily solved this bug.
